File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int biggestSum = -10000000;
        int curSum = 0;
        if (nums.length == 1) return nums[0];
     

        for (int i = 0; i < nums.length; i++){
            
            curSum += nums[i];

            if (curSum > biggestSum){
                biggestSum = curSum;
            }

            if (curSum < 0){
                curSum = 0;
                continue;
            }


            
            
        }
        return biggestSum;

    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        biggestSum = -10000000
        curSum = 0

        for num in nums:
            curSum += num
            if curSum > biggestSum:
                biggestSum = curSum
            
            if curSum < 0:
                curSum = 0

        return biggestSum